File: src/lib/parser/aliases_test.ts
import { assertEquals } from "https://deno.land/std@0.204.0/assert/assert_equals.ts"
import { parseAliases } from "./aliases.ts"

Deno.test(function testParseAliases() {
  assertEquals(
    parseAliases({
      name: "aliases",
      lines: ["#comment", "key = value", "lock = c+g+q"],
      startLineNr: 10,
    }),
    new Map([
      ["key", { value: "value", lineNr: 11 }],
      ["lock", { value: "c+g+q", lineNr: 12 }],
    ]),
  )
})

File: internal/parser/aliases_test.go
package parser

import (
	"testing"

	"github.com/elentok/qmkmd/internal/layout"
)

func TestParseAliases(t *testing.T) {
	t.Parallel()

	got := ParseAliases(layout.Block{
		Name:        "aliases",
		Lines:       []string{"#comment", "key = value", "lock = c+g+q"},
		StartLineNr: 10,
	})

	if len(got) != 2 {
		t.Fatalf("expected 2 aliases, got %d", len(got))
	}
	if got["key"] != (layout.Alias{Value: "value", LineNr: 11}) {
		t.Fatalf("unexpected key alias: %+v", got["key"])
	}
	if got["lock"] != (layout.Alias{Value: "c+g+q", LineNr: 12}) {
		t.Fatalf("unexpected lock alias: %+v", got["lock"])
	}
}
